### PR TITLE
refactor(mobile): repositories for album service

### PR DESCRIPTION
**Pull request by fyfrey**
This PR adds repositories for the album service to access the local database.
Removes all direct dependencies on Isar within the album service.

Could only add single unit test so far (because it first needs repositories for OpenAPI interaction to test any other methods of the album service)

### CODE REVIEW

**Review by jrasm91 — APPROVED** [review]
Looks like a great start to me. Basically no real code changes, just separated out the database code. With some tests and future iterations it should make future improvements a lot easier.

File: mobile/lib/interfaces/album.interface.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    String name, {
+    bool? shared,
+    bool? remote,
+  });
+  Future<Album> update(Album album);
+  Future<void> delete(Album album);
+  Future<List<Album>> getAll({bool? shared});
+  Future<void> removeUsers(Album album, List<User> users);
+  Future<void> addAssets(Album album, List<Asset> assets);
+  Future<void> removeAssets(Album album, List<Asset> assets);
```
[the remark this answers is not in the document]
Comment (fyfrey) [reply mobile/lib/interfaces/album.interface.dart:19]: currently that more convenient as the database requires the full object to perform such actions

File: mobile/lib/interfaces/asset.interface.dart [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import 'package:immich_mobile/entities/album.entity.dart';
+import 'package:immich_mobile/entities/asset.entity.dart';
+import 'package:immich_mobile/entities/user.entity.dart';
+
+abstract interface class IAssetRepository {
+  Future<List<Asset>> getByAlbumWithOwnerUnequal(Album album, User user);
```
[the remark this answers is not in the document]
Comment (fyfrey) [reply mobile/lib/interfaces/asset.interface.dart]: this should actually return the `Asset`s belonging to a specific album but not owned by user x. I've modified the signature; it's now easier to read

File: mobile/lib/interfaces/album.interface.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (jrasm91) [reply mobile/lib/interfaces/album.interface.dart:19]: Ok, sounds good.

**Comment by jrasm91** [comment]
In the server we try to avoid having services depend on other services. Common code used by two is extracted out to a utility file or class, and then testing only has to inject repository mocks. Do you think we can eventually get to that here as well?

**Comment by fyfrey** [comment]
I think we can move in that direction with mobile as well. It will certainly take a few more PRs but it's doable (maybe we need to rename the sync service to a repository :-P)

**Comment by jrasm91** [comment]
@fyfrey feel free to merge these refactor PRs after I give an approval.

**Pull request merged**